components/movementsensor/gpsutils/vrs.go

```diff
@@ -66,3 +82,50 @@ func HasVRSStream(sourceTable *Sourcetable, mountPoint string) (bool, error) {
 
 	return stream.Nmea, nil
 }
+
+// Close closes the VRS connection and any other background threads.
+func (vrs *VRS) Close() error {
+	vrs.cancelFunc()
+	vrs.activeBackgroundWorkers.Wait()
+	return vrs.Conn.Close()
+}
+
+// StartGGAThread starts a thread that writes GGA messages to the VRS.
+func (vrs *VRS) StartGGAThread(ggaFunc func() (string, error)) {
+	vrs.activeBackgroundWorkers.Add(1)
+	go func() {
+		defer vrs.activeBackgroundWorkers.Done()
+		ticker := time.NewTicker(time.Duration(1000.*20) * time.Millisecond)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-vrs.cancelCtx.Done():
+				return
+			case <-ticker.C:
+				// We currently only write the GGA message when we try to reconnect to VRS. Some documentation for VRS states that we
+				// should try to send a GGA message every 5-60 seconds, but more testing is needed to determine if that is required.
+				// get the GGA message from cached data
+				ggaMessage, err := ggaFunc()
+				if err != nil {
+					vrs.logger.Error("Failed to get GGA message: ", err)
+					continue
+				}
+
+				vrs.logger.Debugf("Writing GGA message: %v\n", ggaMessage)
+
+				_, err = vrs.ReaderWriter.WriteString(ggaMessage)
+				if err != nil {
+					vrs.logger.Error("Failed to send NMEA data:", err)
+					continue
+				}
+
+				err = vrs.ReaderWriter.Flush()
```

Comment:
These two paragraphs are copied in gpsrtk.go. Make them a separate function that is called from 2 places.